test/FastTests/Issues/RavenDB-6451.cs

```diff
@@ -0,0 +1,143 @@
+                (from type in referenceAssemblies.SelectMany(x => x.ExportedTypes)
+                    where typeof(Exception).IsAssignableFrom(type) &&
+                          (type.Namespace.StartsWith("Raven") ||
+                           type.Namespace.StartsWith("Voron") ||
+                           type.Namespace.StartsWith("Sparrow"))
+                    select type).ToArray();
+
+            foreach (var t in exceptionTypes)
+            {
+                try
+                {
+                    if(t == typeof(DocumentConflictException))
+                        Debugger.Break();
+                    Assert.False(HasInvalidProperties(t), $"The type {t.FullName} should not have pointer or BlittableXXX properties");
+                }
+                catch (Exception e)
+                {
+                    Console.WriteLine(e);
+                    throw;
+                }
+            }
+        }
+
+        public bool HasInvalidProperties(Type t)
+        {
+            var properties = t.GetProperties();
+            foreach (var p in properties)
+            {
+                if (p.PropertyType.IsPointer)
+                    return true;
+
+                if (p.PropertyType.Name.Contains("Blittable"))
+                    return true;
+
+                if (p.PropertyType.IsArray ||
+                    p.PropertyType.Namespace.StartsWith("System") ||
+                    p.PropertyType.GetTypeInfo().IsEnum ||
+                    p.PropertyType.GetTypeInfo().IsPrimitive)
+                {
+                    continue;
+                }
+
+                if (p.PropertyType.GetTypeInfo().IsPrimitive == false &&
+                    IsIEnumerable(p.PropertyType) == false &&
+                    p.PropertyType.GetTypeInfo().IsInterface == false &&
+                    p.PropertyType.GetTypeInfo().ContainsGenericParameters == false &&
+                    typeof(IEnumerable).IsAssignableFrom(p.PropertyType) == false &&
+                    HasInvalidProperties(p.PropertyType))
+                {
+                    return true;
+                }
+            }        
+
+            return false;
+        }
+
+        public bool IsIEnumerable(Type t)
+        {
+            if (t.IsConstructedGenericType)
+            {
+                var genericType = t.GetGenericTypeDefinition();
+                if(genericType.GetInterfaces().Any(x => x.IsConstructedGenericType && 
+                                                        x.GetGenericTypeDefinition() == typeof(IEnumerable<>)))                    
+                    return true;
+            }
+            return false;
+        }
+
+        //slightly modified version from to http://www.michael-whelan.net/replacing-appdomain-in-dotnet-core/
+        public static IEnumerable<Assembly> GetRuntimeNonMicrosoftAssemblies()
+        {
+            var assemblies = new List<Assembly>();
+            var dependencies = DependencyContext.Default.RuntimeLibraries;
```

Comment:
Maybe you can use same technique as in here https://github.com/ravendb/ravendb/blob/v4.0/test/SlowTests/Tests/NoNonDisposableTests.cs
and avoid additional dependency to 'Microsoft.Extensions.DependencyModel' ?